### PR TITLE
Fix golangci-lint errors.

### PR DESCRIPTION
**Pull request by kirkeby**
Fixes complaints from golang-ci-lint, by ignoring error returns in places they don't matter, nolint'ing test-code because that's not worth it IMO, and fixing the rest of the complaints.

### CODE REVIEW

**Comment by klauspost** [comment]
Sidenote - seems like the recommended way is for actions to be pinned to commit hashes: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

File: api/services.go [hunk 1]
```diff
@@ -30,6 +30,9 @@ var (
 )
 
 func StartServices(logger goa.LogAdapter, ctx context.Context, err error) {
+	if err != nil {
```
Comment (kirkeby) [inline api/services.go:33]: Why u even here, `err`?

File: msgppool.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ var writerMsgpPool = sync.Pool{
 	},
 }
 
+//nolint - versionExt is unused, but someone apparently thought it should be here anyway?
```
Comment (kirkeby) [inline msgppool.go:38]: Just remove versionExt maybe?

File: api/lists.go [hunk 3]
```diff
@@ -130,7 +130,7 @@ func (c *ListsController) GetPercentile(ctx *app.GetPercentileListsContext) erro
 	if !ok {
 		return ctx.NotFound(goa.ErrNotFound("not_found", "list_id", ctx.ListID))
 	}
-	percentile := float64(50.0)
+	var percentile float64
```
Comment (eikmadsen) [inline api/lists.go:133]: Maybe something like this, now that we are touching the code
```
percentile, err2 := strconv.ParseFloat(ctx.FromTop, 64); 
if err2 != nil {
		return ctx.BadRequest(goa.InvalidParamTypeError("from_top", ctx.FromTop, "number"))	
}
```

File: api/services.go (hunk 1 above)
Comment (klauspost) [reply api/services.go:33]: Good Q. Just kill it.

File: blobstore/lazysaver.go [hunk 4]
```diff
@@ -681,5 +681,6 @@ func putDataBufferChunk(p []byte) {
 	if size >= 1<<20 {
 		i = 10
 	}
+	//nolint - don't tell me what to make pointerlike
```
[the remark this answers is not in the document]
Comment (klauspost) [reply blobstore/lazysaver.go:684]: Agree.

File: msgppool.go (hunk 2 above)
Comment (eikmadsen) [reply msgppool.go:38]: sounds good to me.

**Review by eikmadsen — CHANGES_REQUESTED** [review]
minor comments :muscle:

File: api/lists.go (hunk 3 above)
Comment (kirkeby) [reply api/lists.go]: Didn't seem to be any reason it was named `err2`, so I changed that to `err` while I was cleaning up

**Comment by kirkeby** [comment]
> Sidenote - seems like the recommended way is for actions to be pinned to commit hashes: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

I pinned the golangci action to a SHA, it doesn't seem worth it to pin the official actions, if they're compromised all of GH probably is AFAICT.

**Review by eikmadsen — APPROVED** [review]
LGTM, nice work @kirkeby

**Pull request merged**